definitions/checks/system_registration.rb

```diff
@@ -0,0 +1,29 @@
+class Checks::SystemRegistration < ForemanMaintain::Check
+  metadata do
+    label :verify_self_registration
+    description 'Check whether system is self-registered or not'
+    tags :default
+
+    confine do
+      file_exists?('/etc/rhsm/rhsm.conf')
+    end
+  end
+
+  def run
+    if system_is_self_registerd?
+      raise ForemanMaintain::Error::Warn, 'System is self registered'
+    else
+      puts 'System is not self registered'
+    end
+  end
+
+  private
+
+  def system_is_self_registerd?
+    rhsm_hostname.include?(hostname)
```

Comment:
Aah. Good catch. I did a split on ` = ` later checked for `hostname` in the derived array.